packages/clerk-js/src/core/clerk.ts

```diff
@@ -502,7 +503,7 @@ export default class Clerk implements ClerkInterface {
       session = (this.client.sessions.find(x => x.id === session) as ActiveSessionResource) || null;
     }
 
-    let newSession = session === undefined ? this.session : session;
+    let newSession = session ? session : this.session || null;
```

Comment:
This change might look innocent but I am sure it's not. In ClerkJS we have three states for sessions

A. Session is set -> Signed in
B. Null -> Signed out
C. Undefined -> Transitioning from signed out to signed in 

That is this change might have negative side effects.

Reply:
I was influenced by the commend under this. We used to have an `as ActiveSessionResource | null` there but we removed it by accident (we didn't see it caused a typescript error.

Reply:
https://github.com/clerkinc/javascript/pull/1042/files/ce824451e6cb28619825e3fcefbc154e90695ce5#diff-1952d6b8ac4b5495e1dba57c32bff9ce72d43f83595c1bc8532a76cf31feb753R508

Reply:
This is causing the error to fail anyway, so we'll take a closer look!

Reply:
I wouldn't change this line of code. 😅

Reply:
As everyone said, these 2 lines are not equivalent unfortunately! 
We only want to fallback to `this.session` when `session` is undefined, as `undefined` and `null` have a significant meaning for this flow. This is not your fault of course, we need to revisit the `undefined / null / session` approach

Reply:
Should we add a comment highlighting A,B,C as Sokratis mentioned ?